h2o-k8s/src/main/java/water/k8s/KubernetesEmbeddedConfig.java

```diff
@@ -0,0 +1,57 @@
+package water.k8s;
+
+import water.H2O;
+import water.init.AbstractEmbeddedH2OConfig;
+import water.util.Log;
+
+import java.net.InetAddress;
+import java.util.Collection;
+
+public class KubernetesEmbeddedConfig extends AbstractEmbeddedH2OConfig {
+
+    private final String flatfile;
+
+    public KubernetesEmbeddedConfig(final Collection<String> nodeIPs) {
+        this.flatfile = writeFlatFile(nodeIPs);
+    }
+
+    private String writeFlatFile(final Collection<String> nodeIPs) {
+        final StringBuilder flatFileBuilder = new StringBuilder();
+
+        nodeIPs.forEach(nodeIP -> {
+            flatFileBuilder.append(nodeIP);
+            flatFileBuilder.append(":");
+            flatFileBuilder.append(H2O.H2O_DEFAULT_PORT); // All pods are expected to utilize the default H2O port
+            flatFileBuilder.append("\n");
+        });
+
+        return flatFileBuilder.toString();
+    }
+
+    @Override
+    public void notifyAboutEmbeddedWebServerIpPort(InetAddress ip, int port) {
+    }
+
+    @Override
+    public void notifyAboutCloudSize(InetAddress ip, int port, InetAddress leaderIp, int leaderPort, int size) {
+        Log.info(String.format("Created cluster of size %d, leader node IP is '%s'", size, leaderIp.toString()));
+    }
+
+    @Override
+    public boolean providesFlatfile() {
+        return true;
+    }
+
+    @Override
+    public String fetchFlatfile() {
+        return flatfile;
+    }
+
+    @Override
+    public void exit(int status) {
```

Comment:
Is this correct?

Reply:
No actions required on exit. H2O process will be terminated, as well as the docker image. No other resources should be deleted. The services/ingresses/routes still live, ready to serve when the desired state of the underlying cluster of H2O pods goes live again.

If those are temporary and should be deleted with H2O, then this is the responsibility of the orechestration software, not H2O's.

E.g. when healthchecks are implemented (we should maybe do it in this PR), K8S `StatefulSet` will restart the whole cluster of H2O pods once any (even one) node fails - that's the point of having a StatefulSet, stateless applications can be restarted independently of each other and also load balanced. Once the cluster goes up, all the services/ingresses/routes/volumes etc. attached to it will continue to serve their purpose.

From H2O's perspective, do we miss something ? Maybe we should at least log some kind of message about termination ?